pkg/webhook/default_server/uniteddeployment/validating/uniteddeployment_validation.go

```diff
@@ -0,0 +1,166 @@
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("template", "statefulSet"), spec.Template.StatefulSet, "only support one kind of subset"))
+	}
+
+	selector, err := metav1.LabelSelectorAsSelector(spec.Selector)
+	if err != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("selector"), spec.Selector, ""))
+	} else {
+		if spec.Template.StatefulSet != nil {
+			template := spec.Template.StatefulSet.Spec.Template
+			coreTemplate, err := convertPodTemplateSpec(&template)
+			if err != nil {
+				allErrs = append(allErrs, field.Invalid(fldPath.Root(), spec.Template, fmt.Sprintf("Convert_v1_PodTemplateSpec_To_core_PodTemplateSpec failed: %v", err)))
+				return allErrs
+			}
+			allErrs = append(allErrs, appsvalidation.ValidatePodTemplateSpecForStatefulSet(coreTemplate, selector, fldPath.Child("template", "statefulSet"))...)
+		}
+		if spec.Template.ReplicaSet != nil {
+			template := spec.Template.ReplicaSet.Spec.Template
+			coreTemplate, err := convertPodTemplateSpec(&template)
+			if err != nil {
+				allErrs = append(allErrs, field.Invalid(fldPath.Root(), spec.Template, fmt.Sprintf("Convert_v1_PodTemplateSpec_To_core_PodTemplateSpec failed: %v", err)))
+				return allErrs
+			}
+			allErrs = append(allErrs, appsvalidation.ValidatePodTemplateSpecForReplicaSet(coreTemplate, selector, *spec.Replicas, fldPath.Child("template", "replicasSet"))...)
+		}
+	}
+
+	var sumReplicas int32
+	count := 0
+	for _, subset := range spec.Topology.Subsets {
+		if len(subset.Name) == 0 {
+			allErrs = append(allErrs, field.Required(fldPath.Child("topology", "subset", "name"), ""))
+		}
+
+		if errs := apimachineryvalidation.NameIsDNSLabel(subset.Name, false); len(errs) > 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("topology", "subset", "name"), subset.Name, fmt.Sprintf("invalid subset name %s", strings.Join(errs, ", "))))
+		}
+
+		if spec.Replicas == nil || subset.Replicas == nil {
+			continue
+		}
+
+		replicas, err := parseUnitReplicas(*spec.Replicas, *subset.Replicas)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("topology", "subset", "replicas"), subset.Replicas, fmt.Sprintf("invalid replicas %s", subset.Replicas.String())))
+		} else {
+			sumReplicas += replicas
+			count++
+		}
+	}
+
+	// summary of subset replicas may be less than uniteddployment replicas
+	if sumReplicas > *spec.Replicas {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("topology", "subset"), sumReplicas, fmt.Sprintf("summary of indicated subset replicas %d should not be greater than UnitedDeployment replicas %d", sumReplicas, *spec.Replicas)))
+	}
+
+	if count > 0 && count == len(spec.Topology.Subsets) && sumReplicas != *spec.Replicas {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("topology", "subset"), sumReplicas, fmt.Sprintf("if replicas of all subsets are provided, the summary of indicated subset replicas %d should equals UnitedDeployment replicas %d", sumReplicas, *spec.Replicas)))
```

Comment:
summary -> sum
should equals -> should equal or should be equal to

Reply:
done